### PR TITLE
Convert last 6 mojo tests to use subtests.  #1520

### PR DESCRIPTION
**Pull request by spazm**
### Summary
Convert 6 tests to use subtests.

### Motivation
Requested in #1520 

### References
See #1520

### CODE REVIEW

File: t/mojo/reactor_poll.t [hunk 1]
```diff
@@ -7,332 +7,374 @@ use IO::Socket::INET;
...
+  is ref Mojo::Reactor::Poll->new, 'Mojo::Reactor::Poll', 'right object';
+  undef $reactor;
+  is ref Mojo::Reactor::Poll->new, 'Mojo::Reactor::Poll', 'right object';
+  use_ok 'Mojo::IOLoop';
+  $reactor = Mojo::IOLoop->singleton->reactor;
+  is ref $reactor, 'Mojo::Reactor::Poll', 'right object';
+
+};
+
+my $reactor = Mojo::IOLoop->singleton->reactor;
```
Comment (kraih) [inline t/mojo/reactor_poll.t:22]: This seems like completely new code that changes what the test does.

File: t/mojo/reactor_poll.t [hunk 2]
```diff
@@ -7,332 +7,374 @@ use IO::Socket::INET;
...
+subtest "Connect" => sub {
+  $client = IO::Socket::INET->new(PeerAddr => '127.0.0.1', PeerPort => $port);
+  $reactor->timer(1 => sub { shift->stop });
+  $reactor->start;
+  ok $readable, 'handle is readable';
+  ok !$writable, 'handle is not writable';
+
+};
+
+my $server;
```
Comment (kraih) [inline t/mojo/reactor_poll.t:58]: Variable outside of subtest.

File: t/mojo/reactor_poll.t [hunk 3]
```diff
@@ -7,332 +7,374 @@ use IO::Socket::INET;
...
+  $reactor->start;
+  ok $readable, 'handle is readable again';
+  ok $writable, 'handle is writable again';
+  ok !$timer, 'timer was not triggered';
+  ok $recurring, 'recurring was triggered again';
+  ok $next_tick, 'next tick was triggered';
+
+};
+
+my $reactor2;
```
Comment (kraih) [inline t/mojo/reactor_poll.t:156]: Variable outside of subtest.

File: t/mojo/user_agent_tls.t [hunk 4]
```diff
@@ -27,46 +27,56 @@ my $listen
...
-is $tx->res->body, 'works!', 'right content';
-
-# Valid certificates (env)
-$ua = Mojo::UserAgent->new(ioloop => $ua->ioloop);
-{
-  local $ENV{MOJO_CA_FILE}   = 't/mojo/certs/ca.crt';
-  local $ENV{MOJO_CERT_FILE} = 't/mojo/certs/client.crt';
-  local $ENV{MOJO_KEY_FILE}  = 't/mojo/certs/client.key';
-  local $ENV{MOJO_INSECURE}  = 0;
+my $ua;
```
Comment (kraih) [inline t/mojo/user_agent_tls.t:30]: Variable outside of subtest.

**Review by kraih — CHANGES_REQUESTED** [review]
Some tests are fine and could be merged, but others have scoping issues and require more attention.

**Comment by kraih** [comment]
I'm sorry, but i've decided to close these PRs, because they contain too many converted tests at once. And we don't have the review capacity to re-review the whole thing after every scoping issue fix.

File: t/mojo/reactor_poll.t (hunk 1 above)
Comment (spazm) [reply t/mojo/reactor_poll.t:22]: These tests had far too much shared code, which is why so much of it had to be moved out of the subtests.  I was hoping to discuss how to better compartmentalize the tests.  

But then I got into a fugue state of converting more and more -- only to find I was breaking the tests when run on integration, as I was missing local ENV vars to run all the tests.  so it goes.

File: t/mojo/reactor_poll.t (hunk 2 above)
Comment (spazm) [reply t/mojo/reactor_poll.t:58]: variable shared between tests and moved out of subtest.  How best to isolate?

File: t/mojo/user_agent_tls.t (hunk 4 above)
Comment (spazm) [reply t/mojo/user_agent_tls.t:30]: variable shared between tests and moved out of subtest.  How best to isolate?

**Comment by spazm** [comment]
Updated PR to remove the tests that did not convert cleanly.
These three had minimal changes (overview provided in the updated PR description) and will hopefully provide minimal review burden.

Hoping some of the effort can be salvaged to get some small part of this conversion off of your plate.

Please consider re-opening and reviewing.  Sorry for producing a headache rather than the happy joy of found work. 
 Have a groovy day!

**Comment by kraih** [comment]
Thanks for your effort. But from now please just one test per PR. This is just too much to review.

**Comment by kraih** [comment]
I've also updated #1520 with a few additions (one test per PR and skip tests with lots of shared state).

**Comment by kraih** [comment]
If you want to salvage these PRs, limit them to one test file and move everything else in new PRs.

**Pull request closed without merging**